### PR TITLE
Add a material theme switcher on demo site and storybook

### PR DESCRIPTION
**Pull request by gcornut**
# General summary

Add a material theme switcher on component demo pages and story book stories.

# Screenshots

https://user-images.githubusercontent.com/939567/142031460-29c7b5d5-1be7-4148-a475-2d8d40ec4e87.mp4

### CODE REVIEW

File: packages/site-demo/src/components/layout/Main.tsx [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-import { Match } from '@reach/router';
-import React from 'react';
+import { MaterialThemeSwitcher } from '@lumx/react/utils/MaterialThemeSwitcher';
 
 export const Main: React.FC = ({ children }) => (
     <main className="main">
         <div className="main__wrapper">
             <div className="main-header">
-                <Match path="/product/components/*">{({ match }) => match && <LumxVersion />}</Match>
+                <Match path="/product/components/*">
```
Comment (FlorentDinet) [inline packages/site-demo/src/components/layout/Main.tsx:11]: Why not target everything on "product" as the theme has implications on foundations (fonts), also patterns which use components, and tokens / utilities that may vary?

File: packages/site-demo/src/style/layout/main/_main.scss [hunk 2]
```diff
@@ -20,6 +20,8 @@
     position: absolute;
     top: 0;
     right: 0;
+    display: flex;
```
Comment (FlorentDinet) [inline packages/site-demo/src/style/layout/main/_main.scss:23]: I'd add `align-items: center;` to align switch and version button vertically

**Comment by FlorentDinet** [comment]
I'd remove the customization of the switch label font as it looks like not aligned with the switch.
Capture d’écran 2021-11-17 à 14 47 23

And use default switch appearance
Capture d’écran 2021-11-17 à 14 48 56

**Review by juanigalan91 — APPROVED** [review]

**Pull request merged**